pkg/registry/apis/provisioning/repository/github/impl.go

```diff
@@ -75,10 +74,6 @@ const (
 )
 
 func (r *githubClient) GetContents(ctx context.Context, owner, repository, path, ref string) (fileContents RepositoryContent, dirContents []RepositoryContent, err error) {
-	if strings.Contains(path, "..") {
```

Comment:
I don't think we have to check this deep. I think it's good enough to check on entry points of the system (e.g. API, webhook events, etc).

Reply:
I think it's a good idea to do, as the GH client also does so. I'd rather be a little extra defensive with small measures than slip something through :)